cpp/daal/src/algorithms/linear_model/oneapi/linear_model_train_normeq_update_oneapi_impl.i

```diff
@@ -82,32 +80,33 @@ services::Status UpdateKernelOneAPI<algorithmFPType>::compute(NumericTable & xTa
     if (interceptFlag)
     {
         const TypeIds::Id idType = TypeIds::id<algorithmFPType>();
-
+        DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nBetasIntercept, nCols);
+        DAAL_OVERFLOW_CHECK_BY_ADDING(size_t, nBetasIntercept, (nBetasIntercept * nCols));
+        DAAL_ASSERT(xtxBuff.size() >= (nBetasIntercept + nBetasIntercept * nCols));
         sumXBuf = xtxBuff.getSubBuffer(nBetasIntercept * nCols, nBetasIntercept, status);
         DAAL_CHECK_STATUS_VAR(status);
 
         UniversalBuffer sumYBufTmp = context.allocate(idType, nResponses, status);
         DAAL_CHECK_STATUS_VAR(status);
+
         sumYBuf = sumYBufTmp.get<algorithmFPType>();
-        context.fill(sumYBuf, 0.0, status);
+        context.fill(sumYBuf, algorithmFPType(0), status);
         DAAL_CHECK_STATUS_VAR(status);
 
         UniversalBuffer onesBufTmp = context.allocate(idType, nRowsPerBlock, status);
         DAAL_CHECK_STATUS_VAR(status);
         onesBuf = onesBufTmp.get<algorithmFPType>();
 
-        context.fill(onesBuf, 1.0, status);
+        context.fill(onesBuf, algorithmFPType(1), status);
         DAAL_CHECK_STATUS_VAR(status);
     }
 
     for (size_t blockIdx = 0; blockIdx < nBlocks; ++blockIdx)
     {
+        DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, blockIdx, nRowsPerBlock);
```

Comment:
Suggestion: move it out of cycle and check nBlocks * nRowsPerBlock

Reply:
I'll add proposed check. But since `blockIdx` is not `const` the existing checks are reasonable

Reply:
It looks excessive for me but it's up to you.